src/dsp_tools/commands/xmlupload/models/namespace_context.py

```diff
@@ -1,7 +1,28 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any
+
 from rdflib.namespace import Namespace
 
 
-def get_default_json_ld_context() -> dict[str, str]:
+@dataclass(frozen=True)
+class JSONLDContext:
+    context: dict[str, Any]
+
+    def serialise(self) -> dict[str, Any]:
+        return {"@context": self.context}
+
+
+def get_json_ld_context_for_project(ontos: dict[str, str]) -> JSONLDContext:
+    """Provided a dictionary of ontology names and IRIs, returns a JSON-LD context for the project."""
+    context = _get_default_json_ld_context()
+    project_context = _correct_project_context_namespaces(ontos)
+    context.update(project_context)
+    return JSONLDContext(context)
+
+
+def _get_default_json_ld_context() -> dict[str, str]:
```

Comment:
I am keeping this out of the `JSONLDContext` so we don't have to make a dictionary update, everytime we serialise it.